### PR TITLE
reversing dependency linking order for AIX xlC builds

### PR DESCRIPTION
**Pull request by AnnDinKC**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -104,16 +104,13 @@
     <!-- See also: https://ci.appveyor.com/project/maven-nar/nar-maven-plugin -->
   </ciManagement>
   <distributionManagement>
-    <snapshotRepository>
```
Comment (alex-stark) [inline pom.xml]: revert all these pom changes before submitting to the offical repo

File: src/test/java/com/github/maven_nar/test/TestLinkerVersion.java [hunk 2]
```diff
@@ -1,60 +0,0 @@
-/*
```
Comment (alex-stark) [inline src/test/java/com/github/maven_nar/test/TestLinkerVersion.java]: Why delete this test?

File: pom.xml (hunk 1 above)
Comment (AnnDinKC) [reply pom.xml]: i didn't touch this part. i dunno why it's showing up

File: src/test/java/com/github/maven_nar/test/TestLinkerVersion.java (hunk 2 above)
Comment (AnnDinKC) [reply src/test/java/com/github/maven_nar/test/TestLinkerVersion.java]: it breaks for AIX for some reason, out of the box. i deleted it locally but should not have committed that change.

Comment (alex-stark) [reply src/test/java/com/github/maven_nar/test/TestLinkerVersion.java]: i don't htink we submit a PR with a broken test...

File: src/main/java/com/github/maven_nar/Linker.java [hunk 3]
```diff
@@ -387,17 +387,29 @@ public final LinkerDef getLinker(final AbstractCompileMojo mojo, final CCTask ta
...
 
-    // Add transitive dependencies to the shared library search path if we are on linux, directDepsOnly is enabled, and this is not a static library.
-    if (linkPaths != null && linkPaths.size() > 0 && mojo.getDirectDepsOnly() && os.equals(OS.LINUX) && !type.equals(Library.STATIC)){
+    // Add transitive dependencies to the shared library search path if directDepsOnly is enabled, this is not a static library, and the OS is either Linux or AIX.
+    if (linkPaths != null && linkPaths.size() > 0 && mojo.getDirectDepsOnly() && !type.equals(Library.STATIC) && (os.equals(OS.LINUX) || os.equals(OS.AIX))){
         StringBuilder argStrBuilder = new StringBuilder();
-        argStrBuilder.append("-Wl,-rpath-link,");
+        if (os.equals(OS.LINUX))
+        {
+         argStrBuilder.append("-Wl,-rpath-link,");
```
Comment (garlockchristopher) [inline src/main/java/com/github/maven_nar/Linker.java]: Indentation looks off here.

File: src/main/java/com/github/maven_nar/Linker.java [hunk 4]
```diff
@@ -387,17 +387,29 @@ public final LinkerDef getLinker(final AbstractCompileMojo mojo, final CCTask ta
 
     //if No user preference of dependency library link order is specified then use the Default one nar generate.
     if ((this.narDependencyLibOrder == null) && (narDefaultDependencyLibOrder)) {
-        this.narDependencyLibOrder = mojo.dependencyTreeOrderStr(pushDepsToLowestOrder, mojo.getDirectDepsOnly());
+         if (os.equals(OS.AIX) && (getName(null, null).equals("xlC_r") || getName(null, null).equals("xlC") || getName(null, null).equals("xlc"))){
+            String reverse = new StringBuilder(mojo.dependencyTreeOrderStr(pushDepsToLowestOrder, mojo.getDirectDepsOnly())).reverse().toString();
+            this.narDependencyLibOrder = reverse;
+         } else{
```
Comment (garlockchristopher) [inline src/main/java/com/github/maven_nar/Linker.java]: Add a space between else and {

File: src/test/java/com/github/maven_nar/test/TestLinkerVersion.java (hunk 2 above)
Comment (AnnDinKC) [reply src/test/java/com/github/maven_nar/test/TestLinkerVersion.java]: I don't recall but Adam or Michael did. I'll talk to them but I think we agreed it was not our job to fix non-working tests with maven-nar.

Comment (alex-stark) [reply src/test/java/com/github/maven_nar/test/TestLinkerVersion.java]: do you mean this test never passed, or never passed on AIX?  Presumably it worked in some cases at some point right?

Comment (AnnDinKC) [reply src/test/java/com/github/maven_nar/test/TestLinkerVersion.java]: never passed on AIX.

**Pull request merged**